src/main/java/core/basesyntax/service/operation/SupplyOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private StorageDao storageDao;
+
+    public SupplyOperationHandler() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public Map<String, Integer> getOperation(FruitTransaction transaction) {
+        storageDao.save(transaction.getFruit(),
+                storageDao.getQuantity(transaction.getFruit()) + transaction.getQuantity());
+        return storageDao.getAll();
```

Comment:
The `getOperation` method returns the entire storage map, which might not be necessary if the purpose is only to update the storage. Consider returning a more appropriate type or void if the map is not needed after the operation.